zanata-war/src/main/java/org/zanata/action/ProjectPermissionDialog.java

```
@@ -0,0 +1,250 @@
+
+        for (PersonProjectMemberships.LocaleRoles localeRoles: data.getLocaleRoles()) {
+            if(localeRoles.getLocale().equals(hLocale)) {
+                if (StringUtils.equalsIgnoreCase(role, LocaleRole.Translator.name())) {
+                    localeRoles.setTranslator(checked);
+                } else if (StringUtils.equalsIgnoreCase(role, LocaleRole.Reviewer.name())) {
+                    localeRoles.setReviewer(checked);
+                } else if (StringUtils.equalsIgnoreCase(role, LocaleRole.Coordinator.name())) {
+                    localeRoles.setCoordinator(checked);
+                }
+                return;
+            }
+        }
+
+        // No LocaleRoles for the given locale, so create a new one.
+        List<LocaleRole> roleList = Lists.newArrayList(LocaleRole.valueOf(role));
+        data.addLocaleRoles(hLocale, roleList);
+    }
+
+    /**
+     * Save the permissions selections from permissionDialogData to the database.
+     */
+    public void saveSelections() {
+        if (data == null) {
+            log.error("Tried to save permissionDialogData but it is null");
+            return;
+        }
+
+        project = projectDAO.findById(getProject().getId());
+
+        // Hibernate will have problems working with detached HPerson and HLocale
+        // so they are all attached before any persistence is attempted.
+        HPerson person = personDAO.findById(data.getPerson().getId());
+        data.setPerson(person);
+        for (PersonProjectMemberships.LocaleRoles roles : data.getLocaleRoles()) {
+            roles.setLocale(localeServiceImpl
+                    .getByLocaleId(roles.getLocale().getLocaleId()));
+        }
+
+        final boolean canManageMembers = identity.hasPermission(project, "manage-members");
+        final boolean canManageTransMembers = identity.hasPermission(project, "manage-translation-members");
+        final boolean canChangeAnyMembers = canManageMembers || canManageTransMembers;
```

If a user does not have any of the permission, this method will silently do nothing. I think we should have some feedback.
- [ ] feedback about lack of permission